setup.cfg

```diff
@@ -16,4 +16,19 @@ profile = black
 combine_as_imports = True
 
 [tool:pytest]
-addopts = --cov-report= --cov=starlette --cov=tests -rxXs
+addopts =
+  --cov-report=term-missing:skip-covered
+  --cov=starlette
+  --cov=tests
+  -rxXs
+  --strict-config
+  --strict-markers
+  --cov-config tests/.ignore_lifespan
+  --cov-fail-under=100
```

Comment:
Can we keep the coverage requirement out of this PR? Users can check that by running `scripts/coverage`. I can see how it's nice to have everything in one pytest command, but this then does kind of obsolete `scripts/coverage` and more importantly makes this setup different from other encode projects.

Reply:
this is the only place coverage is checked in CI, I think it's important to enforce coverage in CI.

Reply:
Hrm now I see we skip the coverage script in CI. I agree that coverage should be checked in the CI but I'll need to go check why it was skipped in the other project I copied the scripts from.

I still think this is outside the scope of making pytest stricter, though.